it/controllers/benefits/MileageBenefitAmountControllerISpec.scala

```diff
@@ -0,0 +1,513 @@
+            insertCyaData(employmentUserData(isPrior = true, cyaModel("name", hmrc = true, Some(benefits(fullCarVanFuelModel)))), userRequest)
+            authoriseAgentOrIndividual(user.isAgent)
+            urlGet(url(taxYearEOY), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          import Selectors._
+          import user.commonExpectedResults._
+
+          s"has an OK($OK) status" in {
+            result.status shouldBe OK
+          }
+
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(user.commonExpectedResults.expectedCaption(taxYearEOY))
+          buttonCheck(user.commonExpectedResults.continueButtonText, continueButtonSelector)
+          textOnPageCheck(user.specificExpectedResults.get.expectedParagraphWithPrefill, contentSelector)
+          textOnPageCheck(hintText, hintTestSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck("400", amountField)
+          formPostLinkCheck(continueLink, formSelector)
+          welshToggleCheck(user.isWelsh)
+
+        }
+
+        "redirect" when {
+
+          "mileage question is no" which {
+            lazy val result: WSResponse = {
+              dropEmploymentDB()
+              noUserDataStub(nino, taxYearEOY)
+              insertCyaData(employmentUserData(isPrior = false, cyaModel("name", hmrc = true, Some(benefits(fullCarVanFuelModel.copy(mileageQuestion = Some(false)))))), userRequest)
+              authoriseAgentOrIndividual(user.isAgent)
+              urlGet(url(taxYearEOY), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)), follow = false)
+            }
+
+            s"has an SEE_OTHER($SEE_OTHER) status" in {
+              result.status shouldBe SEE_OTHER
+              result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+            }
+
+          }
+          "mileage question is empty" which {
+            lazy val result: WSResponse = {
+              dropEmploymentDB()
+              noUserDataStub(nino, taxYearEOY)
+              insertCyaData(employmentUserData(isPrior = false, cyaModel("name", hmrc = true, Some(benefits(fullCarVanFuelModel.copy(mileageQuestion = None))))), userRequest)
+              authoriseAgentOrIndividual(user.isAgent)
+              urlGet(url(taxYearEOY), welsh = user.isWelsh, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)), follow = false)
+            }
+
+            s"has an SEE_OTHER($SEE_OTHER) status" in {
+              result.status shouldBe SEE_OTHER
+              result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
```

Comment:
do you want a TODO here because it will go to the mileage question page instead of CYA right?